var/spack/repos/spack_repo/builtin/packages/libmetatensor/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack_repo.builtin.build_systems.cmake import CMakePackage
+from spack.package import *
+
+
+class Libmetatensor(CMakePackage):
+    """Self-describing sparse tensor data format for atomistic machine learning and beyond."""
+
+    homepage = "https://docs.metatensor.org"
+    url = "https://github.com/metatensor/metatensor/releases/download/metatensor-core-v0.1.14/metatensor-core-cxx-0.1.14.tar.gz"
+    git = "https://github.com/metatensor/metatensor.git"
+
+    maintainers("HaoZeke", "luthaf")
+    license("BSD-3-Clause", checked_by="HaoZeke")
+
+    version("0.1.14", sha256="dc6cdd9cf0113e2f012ecf68b81cc7cfc71bef3d2020b41574de8fa403dba646")
+    version("0.1.13", sha256="c735b1050357b2873e2e07ee1e263fc9d45faf07f5ea63b65e70869ca423adb5")
+
+    variant("shared", default=True, description="Build shared library version")
+
+    generator("ninja")
+
+    depends_on("cmake", type="build")
+    depends_on("ninja", type="build")
+    depends_on("rust@1.74.0:", type="build")
+
+    depends_on("cxx", type="test")
+    depends_on("cmake", type="test")
+    depends_on("ninja", type="test")
```

Comment:
```suggestion
    depends_on("c", type="build")
    depends_on("cxx", type="build")
    depends_on("cmake", type="build")
    depends_on("ninja", type="build")
    depends_on("rust@1.74.0:", type="build")
```

The only difference with `build` and `test` is that `test` dependencies are only built when using `spack install --test`. Therefore I thin `build` is sufficient here?

As far as I can tell, [metatensor-core/CMakeLists.txt](https://github.com/metatensor/metatensor/blob/7420d65a486bd0ca121755ee8e2563d17cf6bacc/metatensor-core/CMakeLists.txt#L45) enables both `C` and `CXX`. I think both should also be a build dependency.